onnx/shape_inference/implementation.cc

```diff
@@ -115,6 +115,39 @@ void checkShapesAndTypes(const TypeProto& inferredType, const TypeProto& existin
     fail_type_inference("type case unsupported. existing=", existingTypeCase, " inferred=", inferredTypeCase);
   }
 }
+// TypeProto_Tensor or TypeProto_SparseTensor
+template <typename T>
+void generateSymbolicShape(T inferredType, T* symbolicInferredType, SymbolicShape symbolicShape) {
```

Comment:
Unclear why we need the first two parameters, and what the relationship between them is. It looks like in the calling context they differ only in the "const" being cast? Can't we use a single non-const parameter?

Reply:
Sorry that the function here looks confusing previously. Actually the second argument is a mutable tensor type for updating inferredType after symbolic shape_inference. The first argument is the original inferredType. So I think both of them are needed... To make it clearer, I have changed the variable name and updated some comments here.

Reply:
Updated: Only using mutable_tensor_type() can achieve it. Thank you for the comment!